x-pack/legacy/plugins/siem/server/lib/timeline/routes/export_timelines_route.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Hapi from 'hapi';
+
+import { set as _set } from 'lodash/fp';
+import { RequestHandlerContext } from '../../../../../../../../src/core/server';
+import { GetScopedClients } from '../../../services';
+import { LegacyServices, LegacyRequest } from '../../../types';
+import { ExportTimelineRequest, ExportTimelineResults } from '../types';
+import { timelineSavedObjectType } from '../../../saved_objects';
+import { PinnedEvent } from '../../pinned_event/saved_object';
+import { convertSavedObjectToSavedTimeline } from '../convert_saved_object_to_savedtimeline';
+import { transformRulesToNdjson } from '../../detection_engine/routes/rules/utils';
+import { Note } from '../../note/saved_object';
+import { timelineWithReduxProperties } from '../saved_object';
+import { transformError } from '../../detection_engine/routes/utils';
+import { TIMELINE_EXPORT_URL } from '../../../../common/constants';
+
+import { SavedObjectsClient } from '../../../../../../../../src/legacy/server/kbn_server';
+import {
+  exportTimelinesSchema,
+  exportTimelinesQuerySchema,
+} from './schemas/export_timelines_schema';
+import { FrameworkRequest } from '../../framework';
+
+const getExportTimelineByObjectIds = async (
+  client: Pick<
+    SavedObjectsClient,
+    | 'get'
+    | 'delete'
+    | 'errors'
+    | 'create'
+    | 'bulkCreate'
+    | 'find'
+    | 'bulkGet'
+    | 'update'
+    | 'bulkUpdate'
+  >,
+  request: ExportTimelineRequest & LegacyRequest
+) => {
+  const { timeline } = await getTimelinesFromObjects(client, request);
+
+  const timelinesNdjson = transformRulesToNdjson(timeline);
+  return { timelinesNdjson };
+};
+
+const getTimelinesFromObjects = async (
```

Comment:
1. Line 51-101: I'm thinking whether I should put this into `x-pack/legacy/plugins/siem/server/lib/timeline/saved_object.ts` or not.

2. This is what I'm exporting so far, but not sure how to get `globalNotes ` and  `eventNotes `
```
[ { savedObjectId: 'c918c9e0-5e24-11ea-9f8d-03974963d9d4',
    version: 'WzQxMSwxXQ==',
    columns:
     [ [Object],
       [Object],
       [Object],
       [Object],
       [Object],
       [Object],
       [Object],
       [Object] ],
    dataProviders: [],
    description: 'description',
    eventType: 'raw',
    filters: [],
    kqlMode: 'filter',
    kqlQuery: { filterQuery: [Object] },
    title: 'zeek',
    dateRange: { start: 1582538951145, end: 1582625351145 },
    savedQueryId: null,
    sort: { columnId: '@timestamp', sortDirection: 'desc' },
    created: 1583332267291,
    createdBy: 'elastic',
    updated: 1583335392293,
    updatedBy: 'elastic',
    favorite: [],
    eventIdToNoteIds: [],
    noteIds: [],
    notes: [],
    pinnedEventIds: [ '6HW_eHABMQha2n6bHvQ0' ],
    pinnedEventsSaveObject: [ [Object] ] } ]
```

expecting:
```
 columns
 dataProvider
 title
  description
 …x-pack/legacy/plugins/siem/server/lib/timeline/saved_object_mappings.ts
  etc…
 globalNotes: [], x-pack/legacy/plugins/siem/server/lib/note/saved_object_mappings.ts
 eventNotes: [], x-pack/legacy/plugins/siem/server/lib/note/saved_object_mappings.ts
 pinEventIds: [] x-pack/legacy/plugins/siem/server/lib/pinned_event/saved_object_mappings.ts
```